pkg/stanza/fileconsumer/matcher/internal/finder/finder_test.go

```diff
@@ -251,7 +252,31 @@ func TestFindFilesWithIOErrors(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			files, err := FindFiles(tc.include, []string{})
 			assert.ErrorContains(t, err, tc.failedMsg)
-			assert.Equal(t, tc.expected, files)
+			assert.ElementsMatch(t, tc.expected, files)
 		})
 	}
 }
+
+func BenchmarkFind10kFiles(b *testing.B) {
+	numFiles := 10
+	tmpDir := b.TempDir()
+
+	// Create a bunch of files for benchmarking
+	for i := range numFiles {
+		path := filepath.Join(tmpDir, fmt.Sprintf("log-%05d.log", i))
+		f, err := os.Create(path)
+		require.NoError(b, err)
+		require.NoError(b, f.Close())
+	}
+
+	includeGlobs := []string{
+		filepath.Join(tmpDir, "log-*.log"),
+	}
+
+	excludeGlobs := []string{}
+
+	b.ResetTimer()
+	for range b.N {
+		_, _ = FindFiles(includeGlobs, excludeGlobs)
```

Comment:
This has a high likelihood of being optimised out by the golang compiler, I suggest assigning the value to outside the scope of the loop.

https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go

Reply:
Thanks for the tip, I updated to write to a package level variable and reran the benchmarks. Luckily it doesn't look like it was optimized out here.